### ISSUE TITLE
Make Grml "Ventoy Compatible"

### ISSUE THREAD

**Issue by jkirk** [grml/grml#209]
**Describe the bug**
When booting Grml via Ventoy we can not mount the USB device and also the boot options `autoconfig=` fails.

**Steps To Reproduce**

* Create a Ventoy bootable USB devices, see: https://www.ventoy.net/en/doc_start.html
* Copy the Grml ISO to first partition of the USB devices
* Boot Ventoy, select Grml and add the boot option `autoconfig=Ventoy` (with configuration files in a `config.tbz`)

![screenshot_20240906T174349](https://github.com/user-attachments/assets/4057b8ed-1b87-4d44-b676-9994285da645)

* or: try to mount the USB device (e.g. to create a `config.tbz` with `save-config`)

```sh
root@grml ~ # lsblk -o +LABEL
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS                                         LABEL
loop0    7:0    0   827M  1 loop /usr/lib/live/mount/rootfs/grml64-full_sid.squashfs 
                                 /run/live/rootfs/grml64-full_sid.squashfs           
sda      8:0    0 119.5G  0 disk                                                     
├─sda1   8:1    0 119.5G  0 part                                                     Ventoy
└─sda2   8:2    0    32M  0 part                                                     
sr0     11:0    1  1024M  0 rom      
root@grml ~ # mount /dev/sda1 /media/Ventoy
root@grml ~ # mount /dev/sda1 
mount: /media/Ventoy: fsconfig system call failed: /dev/sda1: Can't open blockdev.
       dmesg(1) may have more information after failed mount system call.
```

**Expected behavior**

Load autoconfig configuration even when booting with Ventoy.

And be able to mount the Ventoy partition (e.g. to save the Grml configuration):

```sh
root@grml ~ # lsblk -o +LABEL    
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS                                         LABEL
loop0    7:0    0   827M  1 loop /usr/lib/live/mount/rootfs/grml64-full_sid.squashfs 
                                 /run/live/rootfs/grml64-full_sid.squashfs           
sda      8:0    0 119.5G  0 disk                                                     
├─sda1   8:1    0 119.5G  0 part /media/Ventoy                                       Ventoy
│                                /media/Ventoy                                       
└─sda2   8:2    0    32M  0 part                                                     
sr0     11:0    1  1024M  0 rom    

root@grml ~ # mount /dev/sda1 /media/Ventoy
root@grml ~ # mount | grep Ventoy
/dev/sda1 on /media/Ventoy type exfat (rw,relatime,fmask=0022,dmask=0022,iocharset=utf8,errors=remount-ro)
```

**Workaround**

Set `Mount Ventoy partition after boot Linux` to `[ 1 ]` in `F5 Tools ---> Temporary Control Settings`.

See: https://www.ventoy.net/en/doc_ctrl_settings.html

When applying this workaround `grml-udev-rebuildfstab` sometimes creates an invalid `/etc/fstab`.
(I could not reproduce the output while writing this issue. I will try to provide an example later.)

There seems to be a race-condition.
@mika It would be nice to have a command to invoke `grml-udev-rebuildfstab` manually.

**Real fix**

Grml needs to detects a Ventoy boot and use the "real" device provided by Ventoy (see "Background" below for details).

According to Ventoy, the following things need to be done:

- [ ] Mark the iso file as Ventoy Compatible, see: https://www.ventoy.net/en/doc_compatible_mark.html
- [ ] Check and get ventoy information after boot
- [ ] Mount the iso file according to ventoy information

See the example script here: https://www.ventoy.net/en/compatible.html + https://www.ventoy.net/download/ventoy.sh

**In which Grml Version did the bug occur?**

Output of `grml-version`:
```sh
root@grml ~ # grml-version 
grml64-full_sid build4636 Release Codename autobuild-build4636 [2024-08-31]
```
### Background

**What is Ventoy?**

See: https://www.ventoy.net

Ventoy is an open source tool to create bootable USB drive for ISO/WIM/IMG/VHD(x)/EFI files.
With ventoy, you don't need to format the disk over and over, you just need to copy the ISO/WIM/IMG/VHD(x)/EFI files to the USB drive and boot them directly.
You can copy many files at a time and ventoy will give you a boot menu to select them

**Tested ISOs**

Grml has been officially tested: https://www.ventoy.net/en/isolist.html

`grml64-full_2022.11.iso` is the last ISO in the [Grml Tested Image List](https://www.ventoy.net/en/distro_iso/grml.html)

We should also consider filing a bug report with our latest Grml release (like this [one](https://github.com/ventoy/Ventoy/issues/2239)).

**What is "Ventoy Compatible"?**

See: https://www.ventoy.net/en/compatible.html

Ventoy will create a virtual disk from the ISO file and boot it. But the virtual disk is only on BIOS (Legacy or UEFI) level.
In most case it works only in bootloader period. Most of the modern OS's kernels will use their driver to access hardware after boot, so the virtual disk will not be visible to them.
In normal case, the OS will search all the hardware storage media (CDROM/USB/HD ...) to find the source medium. 
But with ventoy they will not find such medium because there is no such physical media and they don't know that they were booted from a virtual disk. In order to support that ventoy must do some hook before boot. But the hook is really a hard work because there are so many different OS distros and so many special cases. 

Ventoy saves the information to runtime memory before booting and provides a tool to get the information. In this way, the OS can know that they were booted from a virtual disk created by ventoy and can also know which iso file in which disk is the right image.
Based on that, the OS can find the source medium successfully and continue to boot.

**Rational**

We should really support Ventoy to make it easy for first-time users to use Grml.
Especially when they come from Windows.

**Comment by zeha** [comment]
@jkirk is this still in HighPrio for 2026.Q1 or not?

**Comment by mika** [comment]
> [@jkirk](https://github.com/jkirk) is this still in HighPrio for 2026.Q1 or not?

No feedback/progress since 2024 :-/ - but I'd like to raise this, given the latest bug report we received via mail, as of 2026.04 Grml ISOs don't seem to work in Ventoy at all anymore. Might be related to usage of GRUB also for BIOS boot? I guess we'd need to reach out to Ventoy upstream?

**Comment by zeha** [comment]
> I guess we'd need to reach out to Ventoy upstream?

Yes, I think someone with a Ventoy setup should ping Ventoy upstream with the failure report.
